openai/api_requestor.py

```diff
@@ -145,6 +147,63 @@ def format_app_info(cls, info):
             str += " (%s)" % (info["url"],)
         return str
 
+    def __check_polling_response(self, response: OpenAIResponse, predicate: Callable[[OpenAIResponse], bool]):
+        if predicate(response):
+            message, code = (None, None) if 'error' not in response.data else (
+                response.data['error']['message'] if 'message' in response.data['error'] else None,
+                response.data['error']['code'] if 'code' in response.data['error'] else None
+            )
+            message = message if message is not None else 'Operation failed'
+            raise error.OpenAIError(message=message, code=code)
+
+    def poll(
```

Comment:
Do we want to expose this as a public API?

Reply:
What about moving this to the `api_resource` in question? I don't think the `ApiRequestor` is responsible for multi-network call orchestrations - that feels more like a per-resource choice.

Reply:
we probably don't want to expose any of the LRO behavior as public in this library, in that sense and given that the current scope is only for images, do you think we should move it there instead of a more generalized place?

Reply:
If we want the method to be internal/not part of the public API surface area, we should prefix the name with a single underscore (e.g. `_poll`)

Reply:
I think we can default `method`, `until` and `failed`. Azure's API guidelines have well-defined terminal states, and we've been using those in codegen for a long time. Similarly, the method should be `GET` for all polling operations. 

We should also make all optional parameters keyword-only.

Reply:
fixed